src/launch/linux.rs

```diff
@@ -83,166 +79,84 @@ impl From<i32> for TdxError {
-    /// features available for use by the guest TD.
-    pub xfam_fixed0: u64,
-
-    /// Bitmap where if any certain bit is 1, it must be 1 in any TD's XFAM.
-    /// XFAM (eXtended Features Available Mask) determines the set of extended
-    /// features available for use by the guest TD.
-    pub xfam_fixed1: u64,
-
-    /// Supported Guest Physical Address Width
-    pub supported_gpaw: u32,
-
-    /// Padding space. Ignored
-    _padding: u32,
-
-    /// Reserved space. Ignored.
-    _reserved: [u64; 251],
-
-    /// Number of CPUID_CONFIG entries
-    pub nr_cpuid_configs: u32,
-
-    /// Enumeration of the CPUID leaves/sub-leaves that contain bit fields whose
-    /// virtualization by the Intel TDX module is either:
-    ///
-    /// - Directly configurable (CONFIG_DIRECT) by the host VMM
-    /// - Bits that the host VMM may allow to be 1 (ALLOW_DIRECT) and their
-    ///   native value, as returned by the CPU, is 1
-    ///
-    /// Note that the virtualization of many CPUID bit fields not enumerated in
-    /// this list is configurable indirectly via the XFAM and ATTRIBUTES assigned
-    /// to a TD by the host VMM.
-    pub cpuid_configs: [CpuidConfig; NR_CPUID_CONFIGS],
+    /// Number of pages to be mapped
+    pub nr_pages: u64,
 }
 
-impl Default for Capabilities {
-    fn default() -> Self {
-        Self {
-            attrs_fixed0: 0,
-            attrs_fixed1: 0,
-            xfam_fixed0: 0,
-            xfam_fixed1: 0,
-            supported_gpaw: 0,
-            _padding: 0,
-            _reserved: [0; 251],
-
-            nr_cpuid_configs: NR_CPUID_CONFIGS as u32,
-            cpuid_configs: [Default::default(); NR_CPUID_CONFIGS],
-        }
-    }
-}
+#[allow(non_camel_case_types)]
+pub type __u64 = ::std::os::raw::c_ulonglong;
 
-/// TDX specific VM initialization information
-#[derive(Debug)]
 #[repr(C)]
-pub struct InitVm {
-    /// Guest TD attributes
-    pub attributes: u64,
-
-    /// Software-defined ID for non-owner-defined configuration of the guest TD
-    /// (runtime or OS configuration)
-    pub mrconfigid: [u64; 6],
-
-    /// Software-defined ID for the guest TD’s owner
-    pub mrowner: [u64; 6],
-
-    /// Software-defined ID for owner-defined configuration of the guest TD
-    /// (specific to the workload)
-    pub mrownerconfig: [u64; 6],
-
-    /// reserved for future extensibility
-    reserved: [u64; 1004],
-
-    /// direct configuration of CPUID leaves/subleaves virtualization
-    pub cpuid_nent: u32,
-
-    _padding: u32,
-
-    pub cpuid_entries: [kvm_bindings::kvm_cpuid_entry2; 256],
-}
-
-impl InitVm {
-    pub fn new(cpuid_entries: &Vec<kvm_bindings::kvm_cpuid_entry2>) -> Self {
-        Self {
-            cpuid_nent: cpuid_entries.len() as u32,
-            cpuid_entries: cpuid_entries.as_slice().try_into().unwrap(),
-            ..Default::default()
-        }
-    }
+#[derive(Debug)]
+pub struct kvm_tdx_capabilities {
```

Comment:
```suggestion
pub struct Capabilities {
```

Reply:
Just pushed a set of new changes. Hopefully it makes it more clear that I'm using the bindgen bindings from the TDX header. 

I also didn't realize I had poor commit messages, so hopefully those help understand the latest diff.